docs: Improve external etcd documentation wording

mpeiffer opened the pull request:
Changes to be committed:
	modified:   k8s-install-external-etcd.rst
	modified:   requirements_intro.rst

<!-- Description of change -->

Previously, having an available external etcd was not listed under the requirements section in the external etcd documentation. This was addressed by listing "Key-Value store etcd >= 3.1.0 or consul >= 0.6.4" under the requirements section. In addition, the section title "Configure the external etcd" referred to the configuration of Cilium and not an external etcd, thus the title was changed to "Configure Cilium" for clarity.

Fixes: #9401

Signed-off-by: Melissa Peiffer mbp83@nau.edu

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9415)
<!-- Reviewable:end -->

mpeiffer committed:
Listed "Key-Value store etcd >= 3.1.0 or consul >= 0.6.4 under
requirements to clarify an available external etcd is needed.

Renamed "Configure the external etcd" to "Configure Cilium" for clarity.

On branch dev
Changes to be committed:
	modified:   k8s-install-external-etcd.rst
	modified:   requirements_intro.rst

mpeiffer committed:
docs: Improve external etcd documentation wording

Changes to be committed:
	modified:   k8s-install-external-etcd.rst
	modified:   requirements_intro.rst

joestringer commented on Documentation/gettingstarted/requirements_intro.rst:
```diff
@@ -6,6 +6,7 @@ Make sure your Kubernetes environment is meeting the requirements:
 * Kubernetes >= 1.9
 * Linux kernel >= 4.9
 * Kubernetes in CNI mode
+* Key-Value store etcd >= 3.1.0 or consul >= 0.6.4
```
While this improves the docs for the "External etcd" guide, this also now reports such a requirement for the "etcd operator" guide (http://docs.cilium.io/en/latest/gettingstarted/k8s-install-etcd-operator/) which actually requires neither `etcd` nor `consul` to be installed, as that guide will configure `etcd` as part of the guide.

Also, this text is only referred to from etcd guides so the Consul mention is not necessary. That option is already documented here: http://docs.cilium.io/en/latest/install/system_requirements/#summary

```
$ git grep requirements_intro Documentation/
Documentation/gettingstarted/k8s-install-etcd-operator.rst:.. include:: requirements_intro.rst
Documentation/gettingstarted/k8s-install-external-etcd.rst:.. include:: requirements_intro.rst
```

joestringer reviewed (changes requested):
Hi, thanks for the contribution!

As a matter of policy, all commits should be signed off (`git commit -s`): http://docs.cilium.io/en/latest/contributing/contributing/#developer-s-certificate-of-origin

I also commented on the changes below, minor adjustmentments needed.

mpeiffer committed:
Signed-off-by: Melissa Peiffer <mbp83@nau.edu>

Renamed 'Configure External etcd' to 'Configure Cilium' for clarity and
listed an external etcd under the requirements section.

Changes to be committed:
	modified:   k8s-install-external-etcd.rst

mpeiffer committed:
Signed-off-by: Melissa Peiffer <mbp83@nau.edu>

I removed the requirement of etcd or consul from the requirements intro so that
"etcd operator guide" does not report an unecessary requirement.
Instead, I listed the requirement for having etcd under the external etcd
installation guide.

joestringer reviewed (commented):
Looks good, could you just rebase and squash the commits together?

https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history

mpeiffer committed:
Docs: improve external etcd documentation.

Changed name of Configure Extrenal etcd to Configure Cilium for clarity. Added an external etcd as a requirement under the requirements section.

Signed-off-by: Melissa Peiffer <mbp83@nau.edu>

mpeiffer committed:
Merge branch 'dev'
